src/local/experiment.py

```diff
@@ -101,9 +104,9 @@ def eval_and_log(self, metric_fun=r2_score, metric_name='r2'):
         preds_val = self.model.predict(self.x.val)
         preds_test = self.model.predict(self.x.test)
 
-        metric_train = metric_fun(self.y.train, preds_train)
-        metric_val = metric_fun(self.y.val, preds_val)
-        metric_test = metric_fun(self.y.test, preds_test)
+        metric_train = metric_fun(self.y.train, preds_train, sample_weight=self.sw.train)
```

Comment:
Do you think it's worth checking if metric_fun has the 'sample_weight' argument? I'm sure in most cases it does but probably not always. If it doesn't, probably it makes sense to throw an error if weights are not equal

Reply:
I think that checking this is not that easy and for now we do not use any such metric. I think in the future we can do something like that